src/components/Collection/CollectionDetail/CollectionDetail.tsx

```diff
@@ -1,51 +1,178 @@
+    deleteCollectionBookmark(id, bookmarkId).then(() => {
+      onUpdated();
+    });
+  };
+
+  const router = useRouter();
+
+  const sortOptions = [
+    {
+      id: CollectionDetailSortOption.RecentlyAdded,
+      label: t('collection:recently-added'),
+    },
+    {
+      id: CollectionDetailSortOption.VerseKey,
+      label: t('collection:verse-key'),
+    },
+  ];
+
+  const chaptersData = useContext(DataContext);
   const sorter = (
-    <div className={styles.sorter}>
-      {t('collection:recently-added')}
-      <div className={styles.sorterIcon}>
-        <ChevronDownIcon />
-      </div>
-    </div>
+    <CollectionSorter selectedOptionId={sortBy} onChange={onSortByChange} options={sortOptions} />
   );
 
+  const onGoToAyahClicked = (verseKey: string) => {
+    const verseUrl = getChapterWithStartingVerseUrl(verseKey);
+    logButtonClick(
+      // eslint-disable-next-line i18next/no-literal-string
+      `collection_detail_menu_go_to_verse`,
+    );
+    router.push(verseUrl);
+  };
+
   return (
     <div className={styles.container}>
       <div className={styles.header}>
         <div className={styles.title}>{title}</div>
         {sorter}
       </div>
       <div className={styles.collectionItemsContainer}>
-        {collectionItems.map((item) => {
+        {bookmarks.map((bookmark) => {
+          const chapterData = getChapterData(chaptersData, bookmark.key.toString());
+          const verseKey = makeVerseKey(bookmark.key, bookmark.verseNumber);
+          const itemTitle = `${chapterData.transliteratedName} ${toLocalizedVerseKey(
+            verseKey,
+            lang,
+          )}`;
           return (
             <Collapsible
-              title={item.title}
-              key={item.id}
+              title={itemTitle}
+              key={bookmark.id}
               prefix={<ChevronDownIcon />}
-              suffix={<OverflowMenuIcon />}
+              shouldRotatePrefixOnToggle
+              suffix={
+                <PopoverMenu
+                  trigger={
+                    <Button variant={ButtonVariant.Ghost}>
+                      <OverflowMenuIcon />
+                    </Button>
+                  }
+                >
+                  <PopoverMenu.Item onClick={() => onBookmarkItemDeleted(bookmark.id)}>
+                    {t('collection:delete')}
+                  </PopoverMenu.Item>
+                  <PopoverMenu.Item
+                    onClick={() =>
+                      onGoToAyahClicked(makeVerseKey(bookmark.key, bookmark.verseNumber))
+                    }
+                  >
+                    {t('collection:go-to-ayah')}
+                  </PopoverMenu.Item>
+                </PopoverMenu>
+              }
             >
               {({ isOpen }) => {
                 if (!isOpen) return null;
-                return <div className={styles.item}>{item.content}</div>;
+                const chapterId = bookmark.key;
```

Comment:
I think we should extract this to a new component as the file is becoming very bloated.